src/builtindict.js

```diff
@@ -127,7 +134,10 @@ Sk.setupObjects = function (py3) {
         Sk.builtins["xrange"] = new Sk.builtin.func(Sk.builtin.xrange);
         Sk.builtins["StandardError"] = Sk.builtin.Exception;
         Sk.builtins["unicode"] = Sk.builtin.str;
+        Sk.builtins["basestring"] = Sk.builtin.str;
+        delete Sk.builtins["bytes"];
+        delete Sk.builtins["ascii"];
```

Comment:
Python 2's `str`s need to support `decode()` as well as `encode()` (doesn't have to be fast, it just has to work!)
I'd suggest a pretty direct "call `$jsstr()`, barf if it's not a byte-string, turn it into a `Uint8Array`, pass that to the encoding machinery."

Reply:
i think i've fixed it:

```
Sk.builtin.str.$py2decode = new Sk.builtin.func(function (self, encoding, errors) {
    Sk.builtin.pyCheckArgsLen("decode", arguments.length, 1, 3);
    // use a fast javastring constructor path in bytes constructor this.v = new TextEncoder.encode(source); 
    const pyBytes = new Sk.builtin.bytes(self.v); 
    return Sk.builtin.bytes.$decode(pyBytes, encoding, errors);
});
```

Reply:
I think this needs a once over or some python2 tests to check it behaves as expected...